### ISSUE TITLE
pandas.read_csv() encoding issue

### ISSUE THREAD

**Issue by JobCollins** [pandas-dev/pandas#27655]
#### Code Sample

```python
#code snippet
...
    elif filename.endswith('.csv'):
        file_df = pd.read_csv(filed)
....
```
#### Automating pandas csv read into a pandas dataframe.

I am trying to build a Flask app that automates reading of files into a dataframe once they are uploaded. The issue is with CSV files which keep throwing UnicodeDecodeError. Now, setting encoding as follows:

```
    elif filename.endswith('.csv'):
        file_df = pd.read_csv(filed, encoding='utf-8')
```
will only cover for csv files with that particular encoding. I have tried the following script that checks for the encodings for a csv file, especially, once the error is generated:

```
import pkgutil
import encodings
import os

def all_encodings():
    modnames = set([modname for importer, modname, ispkg in pkgutil.walk_packages(
        path=[os.path.dirname(encodings.__file__)], prefix='')])
    aliases = set(encodings.aliases.aliases.values())
    return modnames.union(aliases)

text = b'\xa0'  #given by the error message
for enc in all_encodings():
    try:
        msg = text.decode(enc)
    except Exception:
        continue
    if msg == 'ñ':
        print('Decoding {t} with {enc} is {m}'.format(t=text, enc=enc, m=msg))
    else:
        print('Decoding {t} with {enc} is {m}'.format(t=text, enc=enc, m=msg))
```
and it works well to give the encoding needed for that particular file.

Is there a way pandas can read a csv file and find out the encoding automatically . Or is there a fix to this? Maybe to be a feature (if it does not exist yet) in a future release?

**Comment by realjohnward** [comment]
@JobCollins Can you attach a sample csv file you used as "filename"? 
`    elif filename.endswith('.csv'):
        file_df = pd.read_csv(filed, encoding='utf-8')`

**Comment by TomAugspurger** [comment]
This seems out of scope for pandas. I'd recommend using a library like [chardet](https://pypi.org/project/chardet/) to determine the encoding ahead of time.

**Comment by JobCollins** [comment]
> @JobCollins Can you attach a sample csv file you used as "filename"?
> ` elif filename.endswith('.csv'): file_df = pd.read_csv(filed, encoding='utf-8')`
I tried uploading the csv file but the system does not support the file extension.

**Comment by JobCollins** [comment]
> This seems out of scope for pandas. I'd recommend using a library like [chardet](https://pypi.org/project/chardet/) to determine the encoding ahead of time.

My plan was to create an automated web app that loads and preprocesses data in python. I have checked out [chardet](https://chardet.readthedocs.io/en/latest/usage.html) and it seems like an appropriate utility in this case. I understand this is outside pandas scope but wouldn't it be nice to have this just in case another use encounters the same problem. 
I have helped guys with the same issue on Udacity Forums but I thought the script workaround was not as efficient. Respectfully, I appreciate your contribution and hack all the same.
